### PR TITLE
fix: do not require REGISTRY in create-dev-cluster

### LINKED ISSUE #508: create-dev-cluster shouldn't require $REGISTRY
/kind bug

**What steps did you take and what happened:**
When running `./hack/create-dev-cluster.sh` `$REGISTRY` is required, but since we are loading the image into kind, we really could default it to some sane value.

**What did you expect to happen:**
When running `./hack/create-dev-cluster.sh` `$REGISTRY` would be set to some sane value.

**Environment:**

- cluster-api-provider-azure version: @master
- Kubernetes version: (use `kubectl version`): n/a
- OS (e.g. from `/etc/os-release`): n/a

### PR DESCRIPTION
**Pull request by devigned**
**What this PR does / why we need it**:
To make initial development cluster setup easier, we should default the `REGISTRY` env var.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #508

**Release note**:

```release-note
Default REGISTRY when creating a development cluster using
```

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -153,7 +153,7 @@ started prerequisites section](./getting-started.md#Prerequisites)
 
 #### Creating a dev cluster
 
-The steps below are provided in a convenient script in [hack/create-dev-cluster.sh](hack/create-dev-cluster.sh). Be sure to set `REGISTRY="<container-registry>"`, `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`, `AZURE_SUBSCRIPTION_ID`, and `AZURE_TENANT_ID` before running. Optionally, you can override the different cluster configuration variables. For example, to override the workload cluster name:
+The steps below are provided in a convenient script in [hack/create-dev-cluster.sh](hack/create-dev-cluster.sh). Be sure to set `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`, `AZURE_SUBSCRIPTION_ID`, and `AZURE_TENANT_ID` before running. Optionally, you can override the different cluster configuration variables. For example, to override the workload cluster name:
```
Comment (CecileRobertMichon) [inline docs/development.md:156]: you can remove line 168 as well
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/509/files#diff-1bf07101845de6da98f1c948b82c5ce0R168

Comment (CecileRobertMichon) [reply docs/development.md:156]: or not remove but add a comment next to it saying it's optional maybe, same as the TAG variable

Comment (devigned) [reply docs/development.md:156]: I don't think we can yet, at least not with this change.

Comment (CecileRobertMichon) [reply docs/development.md:156]: oops yeah you're right my bad

**Pull request merged**